src/dbus-objects/addrconf.c

```diff
@@ -636,80 +614,348 @@ ni_objectmodel_addrconf_forward_request(ni_dbus_addrconf_forwarder_t *forwarder,
 /*
  * Forward an addrconf drop call to a supplicant service, such as DHCP or zeroconf
  */
-static dbus_bool_t
-ni_objectmodel_addrconf_forward_release(ni_dbus_addrconf_forwarder_t *forwarder,
-			ni_netdev_t *dev, const ni_dbus_variant_t *dict,
-			ni_dbus_message_t *reply, DBusError *error)
+typedef struct ni_addrconf_drop_args {
+	struct {
+		ni_tristate_t	release;
+	} dhcp;
+} ni_addrconf_drop_args_t;
+
+typedef struct ni_addrconf_drop_forward {
+	ni_addrconf_action_t		action;
+	ni_dbus_addrconf_forwarder_t *	fwdr;
+	ni_addrconf_drop_args_t		args;
+} ni_addrconf_drop_forward_t;
+
+typedef struct ni_addrconf_drop_collect {
+	ni_addrconf_action_t		action;
+	unsigned int			deadline;
+	ni_bool_t			complete;
+} ni_addrconf_drop_collect_t;
+
+static int				ni_addrconf_drop_forward_exec(ni_netdev_t *, ni_addrconf_lease_t *);
+static int				ni_addrconf_drop_collect_exec(ni_netdev_t *, ni_addrconf_lease_t *);
+static ni_addrconf_action_t *		ni_addrconf_drop_collect_complete(ni_addrconf_lease_t *);
+
+static inline void
+ni_addrconf_drop_args_init(const ni_dbus_addrconf_forwarder_t *fwdr, ni_addrconf_drop_args_t *args)
 {
-	ni_uuid_t uuid = NI_UUID_INIT;
-	ni_addrconf_lease_t *lease;
-	int rv = -1;
+	(void)fwdr;
```

Comment:
don't we have a unused macro?

Reply:
We actually don't need the fwdr to decide how to init it -- I've added the parameter for consistency, but we can remove it completely too.

Reply:
removed the unused param in https://github.com/openSUSE/wicked/compare/41a0903..0c9aacc